src/document/index.js

```diff
@@ -0,0 +1,106 @@
+import { omitBy, isArray, isUndefined } from 'lodash';
+
+import service from './service';
+import publicKey from './publicKey';
+import authentication from './authentication';
+import { generateDocument, isEquivalentId } from './utils';
+
+class Document {
+    #content;
+
+    constructor({ content }) {
+        this.#content = {
+            publicKey: [],
+            authentication: [],
+            service: [],
+            ...content,
+        };
+    }
+
+    getContent() {
+        return omitBy(this.#content, (prop) => isUndefined(prop) || (isArray(prop) && prop.length === 0));
+    }
+
+    addPublicKey(props) {
+        props.id = publicKey.createId(this.#content.id, props.id);
+        props.controller = props.controller || this.#content.id;
+
+        publicKey.assert(props, this.#content.publicKey);
+
+        this.#content.publicKey.push(props);
+        this.#refreshUpdated();
```

Comment:
Should this be done here or outside, just once, before publishing?

Reply:
As discussed previously, this is due to `refreshUpdated`  being a private method and not available just before publishing. This seemed the best way to do it.